cmake/externals/sentencepieceproject.cmake

```diff
@@ -8,14 +8,16 @@ FetchContent_GetProperties(spm)
 
 if(NOT spm_POPULATED)
   FetchContent_Populate(spm)
-  # need to patch sentencepiece to use protobuf provided by ort. This is ifdef onnxruntime_BUILD_WEBASSEMBLY
-  # for now but there is no real reason to not use it for all builds.
   # 'git apply' should be in FetchContent_Declare() but that creates issues in sucessive builds so for now it is using execute_process().
-  if(onnxruntime_BUILD_WEBASSEMBLY)
-    set(SPM_PATCH "${PROJECT_SOURCE_DIR}/cmake/externals/sentencepieceproject.patch")
+  foreach(SPM_PATCH IN ITEMS
+          # use protobuf provided by ORT
+          "${PROJECT_SOURCE_DIR}/cmake/externals/sentencepieceproject.protobuf.patch"
+          # fix for iOS build
+          "${PROJECT_SOURCE_DIR}/cmake/externals/sentencepieceproject.ios.patch"
+          )
     message("-- sentencepiece: patching with ${SPM_PATCH}")
     execute_process(COMMAND git apply --ignore-space-change --ignore-whitespace ${SPM_PATCH} WORKING_DIRECTORY ${spm_SOURCE_DIR})
```

Comment:
Please note that "git apply" won't work if neither the working directory nor its ancestor-directories contain ".git". For example, when doing a out of source build.

Reply:
Noted. With the current setup it should work since we specify a git repo in FetchContent_Declare().

Reply:
Yes, git apply is common on FetchContent_declare. And not sure if there is a case that git cannot be used with FetchContent_